quota/etcd/quotaapi/quota_server.go

```diff
@@ -200,3 +201,13 @@ func findByName(name string, cfgs *storagepb.Configs) (*storagepb.Config, bool)
 	}
 	return nil, false
 }
+
+func validateName(name string) error {
+	switch {
+	case name == "":
+		return status.Errorf(codes.InvalidArgument, "name is required")
+	case !storage.IsNameValid(name):
```

Comment:
Seemed odd to me that naming is a storage responsibility. Are you happy with this boundary?